controllers/chat/sendAnonymousMessage.js

```diff
@@ -101,7 +109,43 @@ const sendAnonymousMessage = async (req, res) => {
       });
     }
 
-    const chat = await channel.sendMessage(baseMessage);
+    const chat = await channel.sendMessage(baseMessage, {skip_push: true});
+    let notificationPayload = {
+      title: `New Message from ${senderInfo?.user?.username}`,
+      body: `${message.substring(0, 100)}`
+    };
+    let dataPayload = {
+      channel_id: channelId,
+      user_id: req.userId,
+      messages_id: chat.message.id,
+      message,
+      message_schema: currentMessageType,
+      type: 'message.new',
+      status: 'sent',
+      is_big_message: 'false',
+      attachment: JSON.stringify(attachments) ?? '',
+      created_at: chat.message.created_at,
+      is_annoymous: 'false'
+    };
+    const dataSizeInBytes = Buffer.byteLength(JSON.stringify(dataPayload), 'utf8');
+    const dataSizeInKilobytes = dataSizeInBytes / 1024;
+    if (dataSizeInKilobytes > 4) {
+      dataPayload.is_big_message = 'true';
+      dataPayload.attachment = '';
+    }
+    channelMember.forEach((member) => {
+      let payload = {
+        notification: notificationPayload,
+        data: {
+          ...dataPayload,
+          is_annoymous: member.is_anonymous.toString()
+        }
+      };
+      BetterSocialCore.fcmToken.sendChatNotification(member.user_id, payload, {
+        priority: 'high',
+        content_available: true
+      });
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The notification payload and data payload are created and sent without any validation of their content. This could potentially lead to issues if any of the values are `null` or `undefined`. Also, the `forEach` loop doesn't handle any errors that might occur during the notification sending process. Consider adding error handling inside the loop.
```diff
+ if (!senderInfo?.user?.username || !message || !channelId || !req.userId || !chat.message.id || !currentMessageType || !chat.message.created_at) {
+   throw new Error('One or more required fields are missing for the payload');
+ }
  const chat = await channel.sendMessage(baseMessage, {skip_push: true});
  let notificationPayload = {
    title: `New Message from ${senderInfo?.user?.username}`,
    body: `${message.substring(0, 100)}`
  };
  let dataPayload = {
    channel_id: channelId,
    user_id: req.userId,
    messages_id: chat.message.id,
    message,
    message_schema: currentMessageType,
    type: 'message.new',
    status: 'sent',
    is_big_message: 'false',
    attachment: JSON.stringify(attachments) ?? '',
    created_at: chat.message.created_at,
    is_annoymous: 'false'
  };
  // ...
  channelMember.forEach((member) => {
    let payload = {
      notification: notificationPayload,
      data: {
        ...dataPayload,
        is_annoymous: member.is_anonymous.toString()
      }
    };
+   try {
      BetterSocialCore.fcmToken.sendChatNotification(member.user_id, payload, {
        priority: 'high',
        content_available: true
      });
+   } catch (error) {
+     console.error(`Failed to send notification to user ${member.user_id}:`, error);
+   }
  });
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->